src/interpreter.rs

```diff
@@ -264,7 +264,11 @@ impl<'a, 'b, V: Verifier, C: ContextObject> Interpreter<'a, 'b, V, C> {
             // BPF_ALU class
             ebpf::ADD32_IMM  => self.reg[dst] = (self.reg[dst] as i32).wrapping_add(insn.imm as i32)      as u64,
             ebpf::ADD32_REG  => self.reg[dst] = (self.reg[dst] as i32).wrapping_add(self.reg[src] as i32) as u64,
-            ebpf::SUB32_IMM  => self.reg[dst] = (self.reg[dst] as i32).wrapping_sub(insn.imm as i32)      as u64,
+            ebpf::SUB32_IMM  => if self.executable.get_sbpf_version().enable_neg() {
```

Comment:
I would use a more explicit swap_sub_operands() or something instead of
piggybacking on `enable_neg`. These flags are cheap and having an explicit one
makes the logic easier to follow